src/core/rq/RegionQueryService_Dummy.cpp

```diff
@@ -0,0 +1,58 @@
+#ifdef DUMMY_RQ
+
+#include "RegionQueryService.hpp"
+#include "../design.hpp"
+
+RegionQuery::RegionQuery()
+{
+    m_ws = std::shared_ptr<Design>(new Design);
+}
+
+void RegionQuery::insertObject(IShapePtr shape)
+{
+    m_ws->addObject(shape);
+}
+
+void RegionQuery::removeObject(IShapePtr shape)
+{
+    m_ws->removeObject(shape);
+}
+
+IShapePtr RegionQuery::getShapeUnderPos(const QPoint & point) const
+{
+    for (auto shape : m_ws->getObjects())
+        if (shape->contains(point))
+            return shape;
+
+    return nullptr;
+}
+
+std::vector<IShapePtr> RegionQuery::getShapesUnderRect(const QRect & box) const
+{
+    std::vector<IShapePtr> shapes;
+    // Iterte over all objects in design and check if current shapes point are inside query box
+    // consider it is, if at least 1 point is in that rect
+    for (auto shape : m_ws->getObjects())
+    {
+        bool contains = false;
+
+        for (auto point : shape->getPoints())
```

Comment:
@nagaina shape getPoints() only returns left upper and right bottom points, soo this check is succussed only if one of this pints are inside query rect. please help. : (

In image selection ( rq ) should consider all 1,2 and 3 objects.

<img width="227" alt="image" src="https://github.com/mm-project/qt_painter/assets/127693048/e898a90a-ed26-40f0-a8ca-90d7efcfea69">